spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskService.java

```diff
@@ -145,13 +146,15 @@ public long executeTask(String taskName, Map<String, String> taskDeploymentPrope
 		Resource metadataResource = appRegistration.getMetadataResource();
 
 		TaskExecution taskExecution = taskExecutionRepository.createTaskExecution();
-		taskDefinition = this.updateTaskProperties(taskDefinition, taskExecution);
+		taskDefinition = this.updateTaskProperties(taskDefinition);
 
 		Map<String, String> appDeploymentProperties = extractAppProperties(taskDefinition.getRegisteredAppName(), taskDeploymentProperties);
 		Map<String, String> deployerDeploymentProperties = DeploymentPropertiesUtils.extractAndQualifyDeployerProperties(taskDeploymentProperties, taskDefinition.getRegisteredAppName());
 		AppDefinition revisedDefinition = mergeAndExpandAppProperties(taskDefinition, metadataResource, appDeploymentProperties);
 
-		AppDeploymentRequest request = new AppDeploymentRequest(revisedDefinition, appResource, deployerDeploymentProperties, commandLineArgs);
+		List<String> updatedCmdLineArgs = new ArrayList<>(commandLineArgs);
```

Comment:
What I'm saying is there is no reason for a user to pass that property himself.
And IF we want to protect against this, then we want to apply protection for soooo many others IMO